src/lib/frontend/d_cnf.ml

```diff
@@ -585,11 +585,11 @@ and handle_ty_app ?(update = false) ty_c l =
 let mk_ty_decl (ty_c: DE.ty_cst) =
   match DT.definition ty_c with
   | Some (
-      Adt { cases = [| { cstr = { id_ty; _ } as cstr; dstrs; _ } |]; _ } as adt
+      Adt { cases = [| { cstr = { id_ty; _ } as cstr; dstrs; _ } |]; _ }
     ) ->
     (* Records and adts that only have one case are treated in the same way,
        and considered as records. *)
-    Nest.add_nest [adt];
+    let uid = Uid.of_ty_cst ty_c in
```

Comment:
Why no `Nest.attach_orders` here?

Reply:
Because we never send record values to the ADT theory so we never use the total order.

Reply:
I did this modification. We have an order on all constructors because we check it in `Uid.of_ty_cst` now ;)